csharp/ql/src/semmle/code/csharp/ir/implementation/raw/internal/TranslatedExpr.qll

```diff
@@ -0,0 +1,2432 @@
+    expr instanceof AsExpr and result instanceof Opcode::CheckedConvertOrNull
+  }
+ }
+
+
+private Opcode binaryBitwiseOpcode(BinaryBitwiseOperation expr) {
+  expr instanceof LShiftExpr and result instanceof Opcode::ShiftLeft or
+  expr instanceof RShiftExpr and result instanceof Opcode::ShiftRight or
+  expr instanceof BitwiseAndExpr and result instanceof Opcode::BitAnd or
+  expr instanceof BitwiseOrExpr and result instanceof Opcode::BitOr or
+  expr instanceof BitwiseXorExpr and result instanceof Opcode::BitXor
+}
+
+private Opcode binaryArithmeticOpcode(BinaryArithmeticOperation expr) {
+  expr instanceof AddExpr and result instanceof Opcode::Add or
+  expr instanceof SubExpr and result instanceof Opcode::Sub or
+  expr instanceof MulExpr and result instanceof Opcode::Mul or
+  expr instanceof DivExpr and result instanceof Opcode::Div or
+  expr instanceof RemExpr and result instanceof Opcode::Rem
+}
+
+private Opcode comparisonOpcode(ComparisonOperation expr) {
+  expr instanceof EQExpr and result instanceof Opcode::CompareEQ or
+  expr instanceof NEExpr and result instanceof Opcode::CompareNE or
+  expr instanceof LTExpr and result instanceof Opcode::CompareLT or
+  expr instanceof GTExpr and result instanceof Opcode::CompareGT or
+  expr instanceof LEExpr and result instanceof Opcode::CompareLE or
+  expr instanceof GEExpr and result instanceof Opcode::CompareGE
+}
+
+/**
+ * IR translation of a simple binary operation.
+ */
+class TranslatedBinaryOperation extends TranslatedSingleInstructionExpr {
+  TranslatedBinaryOperation() {
+    expr instanceof BinaryArithmeticOperation or
+    expr instanceof BinaryBitwiseOperation or
+    expr instanceof ComparisonOperation
+  }
+
+  override Instruction getFirstInstruction() {
+    result = getLeftOperand().getFirstInstruction()
+  }
+
+  override final TranslatedElement getChild(int id) {
+    id = 0 and result = getLeftOperand() or
+    id = 1 and result = getRightOperand()
+  }
+
+  override final Instruction getInstructionOperand(InstructionTag tag,
+    OperandTag operandTag) {
+    tag = OnlyInstructionTag() and
+    if swapOperandsOnOp() then (
+      (
+        operandTag instanceof RightOperandTag and
+        result = getLeftOperand().getResult()
+      ) or
+      (
+        operandTag instanceof LeftOperandTag and
+        result = getRightOperand().getResult()
+      )
+    )
+    else (
+      (
+        operandTag instanceof LeftOperandTag and
+        result = getLeftOperand().getResult()
+      ) or
+      (
+        operandTag instanceof RightOperandTag and
+        result = getRightOperand().getResult()
+      )
+    )
+  }
+
+  override Instruction getInstructionSuccessor(InstructionTag tag,
+    EdgeKind kind) {
+    tag = OnlyInstructionTag() and
+    result = getParent().getChildSuccessor(this) and
+    kind instanceof GotoEdge
+  }
+  
+  override Instruction getChildSuccessor(TranslatedElement child) {
+    (
```

Comment:
No, since we don't care in which order the instructions for the operand expressions are generated, we only care about their order as operands of the binary instruction.